changelog/db.changelog-1.1.xml

```diff
@@ -0,0 +1,131 @@
+      <column name="dueDate" type="VARCHAR(10)" />
+      <column name="invoiceNumber" type="VARCHAR(30)" />
+      <column name="marketingYear" type="INTEGER" />
+      <column name="referenceId" type="UUID" /> <!-- interesting 1.23 -->
+      <column name="schedule" type="VARCHAR(3)" />
+      <column name="sitiAgriInvoiceNumber" type="VARCHAR(20)" />
+      <column name="submitted" type="TIMESTAMP" />
+      <column name="value" type="INTEGER" />
+    </createTable>
+  </changeSet>
+  <changeSet author="Marc Templeton" id="4">
+    <createTable tableName="settlements">
+      <column name="settlementId" type="INTEGER" autoIncrement="true">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="settlementsId_pkey" />
+      </column>
+      <column name="paymentRequestId" type="INTEGER" >
+        <constraints nullable="false" />
+      </column>
+      <column name="detail" type="VARCHAR(255)" />
+      <column name="invalid" type="BOOLEAN" defaultValue="false" />
+      <column name="generated" type="BOOLEAN" defaultValue="false" />
+      <column name="frn" type="BIGINT" />
+      <column name="invoiceNumber" type="VARCHAR(30)" />
+      <column name="ledger" type="VARCHAR(2)" />
+      <column name="reference" type="VARCHAR(9)" /> <!-- based on PY1234567 -->
+      <column name="settled" type="BOOLEAN" defaultValue="false" />
+      <column name="settlementDate" type="TIMESTAMP" /> <!-- interesting in pay-pr this (named settled) is used for above bool -->
+      <column name="sourceSystem" type="VARCHAR(50)" />
+    </createTable>
+  </changeSet>
+  <changeSet author="Marc Templeton" id="5">
+    <addForeignKeyConstraint baseColumnNames="paymentRequestId" baseTableName="settlements" constraintName="fk_paymentRequests_paymentRequestId" referencedColumnNames="paymentRequestId" referencedTableName="paymentRequests" />
+  </changeSet>
+  <changeSet author="Marc Templeton" id="6">
+    <createTable tableName="schedules">
+      <column name="scheduleId" type="INTEGER" autoIncrement="true">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="schedulesId_pkey" />
+      </column>
+      <column name="settlementId" type="INTEGER" >
+        <constraints nullable="false" />
+      </column>
+      <column name="completed" type="TIMESTAMP" />
+      <column name="planned" type="TIMESTAMP" />
+      <column name="started" type="TIMESTAMP" />
+    </createTable>
+  </changeSet>
+  <changeSet author="Marc Templeton" id="7">
+    <addForeignKeyConstraint baseColumnNames="settlementId" baseTableName="schedules" constraintName="fk_settlements_settlementId" referencedColumnNames="settlementId" referencedTableName="settlements" />
+  </changeSet>
+  <changeSet author="Marc Templeton" id="8">
+    <createTable tableName="organisations">
+      <column name="sbi" type="INTEGER">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="organisationsId_pkey" />
```

Comment:
One for a healthy debate: this should not integer because it isn't one

Doing arithmetic on `sbi` and/or `frn` yields nothing from the result, i.e. if we take the `frn` 1000003 and multiple it by 2, the output (2000006) has no meaning -- it doesn't point to the customer which is twice as 'good' or 'bigger' or 'more' than the customer we started on (it likely doesn't point to any customer).

However, in contrast, if we take the `areaClaimed` 32.5 and multiple it by 2, the output (65) has meaning -- it's double the land applied for

Reply:
I feel you're trying to start a fight here... 

I don't disagree, but the advantage of it being an integer is for validation of values.  Either through user input for "click here to generate a statement for SBI xxx" or our own processing.  ie SBI is an incrementing number and we know the upper and lower boundaries to help that validation.  Validating it as a string means we'd probably covert it to an integer anyway to work with it.

It's also consistent with the rest of the payment service so we can avoid needing to stringify it before saving it.

I'd keep it as a string but take your point.

Reply:
_Yeeeeeaaaaahh_ I always thought tht would be the conclusion, unfort my 'leading zeros' agrument was clearly thought of when they started `frn` from `10000000` or whatever it is 😆 

Ik you're a string identifier at heart John as below shows vv (screengrabbed it incase the integer identifier ppl forced you to update the comment)

![Screenshot 2022-08-01 at 15 15 11](https://user-images.githubusercontent.com/56440327/182167850-a7c42336-2d03-4f5d-9129-c2e110a9da9f.png)